### PR TITLE
dev: simplify WithOtherFields<T> usage

### PR DESCRIPTION
**Pull request by Gerson2102**
This PR is simplifying the usage of some types in the codebase to improve readability and management.

Time spent on this PR: 45 minutes

Resolves: #1398 

# Pull Request type
Normal

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/providers/eth_provider/blocks.rs [hunk 1]
```diff
@@ -12,6 +12,8 @@ use mongodb::bson::doc;
 use reth_primitives::{BlockId, BlockNumberOrTag};
 use tracing::Instrument;
 
+pub type ExtendedTransaction = WithOtherFields<Transaction>;
+pub type ExtendedBlock = WithOtherFields<Block<ExtendedTransaction>>;
```
Comment (tcoratger) [inline src/providers/eth_provider/blocks.rs:16]: This is exactly the goal of this PR, can you just move these constants to the appropriate files with header documentation each time to understand what these types define:
- `ExtendedTransaction` could be defined in the file where `StoredTransaction` is defined for example.
- `ExtendedBlock` could be defined in the file where `StoredHeader` is defined for example.

Then you can apply the changed to the whole codebase.

Comment (Gerson2102) [reply src/providers/eth_provider/blocks.rs:16]: When you say "defined" do you mean to add for example `ExtendedTransaction` to this file in `src/providers/eth_provider/database/types/transaction.rs`? in `types/ethereum.rs` we are using `StoredTransaction` and `StoredHeader` maybe that file? I'm not getting the "defined" part.

Comment (tcoratger) [reply src/providers/eth_provider/blocks.rs:16]: Defined just means moving `pub type ExtendedTransaction = WithOtherFields<Transaction>;` to this file :)

Comment (Gerson2102) [reply src/providers/eth_provider/blocks.rs:16]: So moving `ExtendedTransaction` to this path `src/providers/eth_provider/database/types/transaction.rs`? And then importing the type where is neccessary.

Comment (tcoratger) [reply src/providers/eth_provider/blocks.rs:16]: Yes correct it makes more sense for this type to be defined in appropriate files

**Comment by Gerson2102** [comment]
I'm done with all ur comments mate. Lmk what u think!

**Review by tcoratger — APPROVED** [review]

**Pull request merged**